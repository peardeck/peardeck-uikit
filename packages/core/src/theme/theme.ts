/**
 * https://styled-system.com/theme-specification
 */
const colorNames = {
  white: "#ffffff",
  black: "#000000",
  gray: "#eeeeee",
  lightGray: "#fafafa",
  midGray: "#777777",
  blue: "#0000aa",
  green: "#00aa00",
};

const colors = {
  ...colorNames,
  background: colorNames.white,
  text: colorNames.black,
  primary: "#29abe2",
  secondary: "#D24949",
  accent: "#f0a",
  pear: "#c9de55",
  muted: colorNames.lightGray,
  border: colorNames.midGray,
};

const space = [0, 4, 8, 16, 32, 64, 128, 256, 512];

export const theme = {
  mediaQueries: {},
  breakpoints: [48, 52, 64].map((value): string => `${value}rem`),
  space,
  colors,
  // colorModes: {
  //   dark: {
  //     text: colorNames.white,
  //     background: colorNames.black,
  //     primary: "#0cf",
  //     secondary: "#f0e",
  //     accent: "#f0a",
  //     pear: "#c9de55",
  //     muted: colorNames.midGray,
  //   },
  //   cyan: {
  //     text: "#023",
  //     background: "#0ff",
  //     primary: "#03c",
  //     secondary: "#01a",
  //     accent: "#f0a",
  //     pear: "#c9de55",
  //     muted: "#0ee",
  //   },
  //   gray: {
  //     text: "#eef",
  //     background: "#333336",
  //     primary: "#09f",
  //     secondary: "#0bf",
  //     accent: "#f0a",
  //     pear: "#c9c9c9",
  //     muted: "#444448",
  //   },
  //   book: {
  //     text: "#322",
  //     background: "#fff9f9",
  //     primary: "#c30",
  //     secondary: "#400",
  //     accent: "#f0a",
  //     pear: "#c9de55",
  //     muted: "#f9f6f6",
  //   },
  // },

  fonts: {
    body: ["ProximaNovaSoft-Regular", "sans-serif"],
    monospace: [
      "Consolas",
      '"Liberation Mono"',
      "Menlo",
      "Courier",
      "monospace",
    ],
  },

  fontSizes: [12, 14, 16, 20, 24, 32, 48, 64, 72],

  fontWeights: {
    light: 300,
    normal: 400,
    medium: 500,
    semibold: 600,
    bold: 700,
    heavy: 800,
  },

  lineHeights: {
    condensedUltra: 1,
    condensed: 1.25,
    default: 1.5,

    body: 1.7,
    title: 1.25,
  },

  letterSpacings: [],

  sizes: [],

  borders: [0, `1px solid ${colorNames.midGray}`],

  borderStyles: [],

  borderWidths: [],

  radii: [0, 4, 8, 16],

  shadows: [
    "none",
    "0 1px 4px 0 rgba(0,0,0,0.27)",
    "0 2px 8px 0 rgba(0,0,0,0.27)",
    "0 4px 16px 0 rgba(0,0,0,0.27)",
    "0 8px 32px 0 rgba(0,0,0,0.27)",
  ],

  zIndices: [],

  buttons: {
    default: {
      bg: "red",
    },
    //   // default: {
    //   //   color: "primary",
    //   //   // border: "1px solid #dce2e9",
    //   //   // color: "#354242",
    //   //   // backgroundColor: "#ffffff",
    //   // },
    //   // primary: {
    //   //   // color: "#000000",
    //   //   // backgroundColor: "#ff0000",
    //   //   // "&:hover": {
    //   //   //   backgroundColor: "#ff9999",
    //   //   // },
    //   // },
    //   // secondary: {
    //   //   // color: "#eeeeee",
    //   //   // backgroundColor: "#0000aa",
    //   // },
  },

  colorStyles: {
    success: {
      color: "#8FA340",
    },
    info: {
      color: "#444546",
    },
    danger: {
      color: "#D24949",
    },
    warning: {
      color: "#F5E98F",
    },
  },

  textStyles: {
    caps: {
      textTransform: "uppercase",
    },
    smallCaps: {
      textTransform: "uppercase",
      letterSpacing: "0.1em", // TODO: Replace with `letterSpacings` value
    },
  },
};

// TODO: https://www.styled-components.com/docs/api#create-a-declarations-file
//
// declare module 'styled-components' {
//   export interface DefaultTheme {
//     borderRadius: string

//     colors: {
//       main: string
//       secondary: string
//     }
//   }
// }
export interface ThemedComponentProps {
  theme?: typeof theme;
}
