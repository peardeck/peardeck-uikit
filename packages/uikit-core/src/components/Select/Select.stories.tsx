import React from "react";
import {
  colorKnobs,
  displayKnobs,
  spaceKnobs,
  typographyKnobs,
} from "../../theme/knobs";
import { Box } from "../Box/Box";
import { Select } from "./Select";

export default {
  component: Select,
  title: "core|Select",
};

export const Default = () => (
  <Box m={2}>
    <Select
      {...typographyKnobs()}
      {...displayKnobs()}
      {...spaceKnobs()}
      {...colorKnobs()}
    >
      <option value="1">Option 1</option>
      <option value="2">Option 2</option>
      <option value="3">Option 3</option>
      <option value="4">Option 4</option>
    </Select>
  </Box>
);
