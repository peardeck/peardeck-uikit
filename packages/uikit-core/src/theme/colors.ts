/**
 * Some standard color scales that can be used for general-purpose
 * styling or as a basis for theme colors.
 * These can be references like: `<Thing color="blue.5" />`
 */
export default {
  black: "#000",
  white: "#fff",
  gray: [
    "#eaeaea",
    "#cfcfcf",
    "#b5b5b5",
    "#9c9c9c",
    "#838383",
    "#6b6b6b",
    "#555555",
    "#3f3f3f",
    "#2a2a2a",
  ],
  blue: [
    "#deeef6",
    "#c3e0ef",
    "#a9d2e8",
    "#8ec4e1",
    "#65aed7",
    "#3087B7",
    "#29739c",
    "#225f81",
    "#1b4b66",
    "#102e3e",
  ],
  green: [
    "#f7ffda",
    "#eaff96",
    "#dcff52",
    "#ceff0e",
    "#a0c900",
    "#6A8500",
    "#4f6300",
    "#344100",
    "#263000",
    "#191f00",
  ],
  yellow: [
    "#ffff9c",
    "#f9f3c1",
    "#f6eda1",
    "#f5e98f",
    "#f1e163",
    "#eedb44",
    "#ebd525",
    "#dac414",
    "#bba811",
    "#9c8c0e",
  ],
  orange: [
    "#ffe1aa",
    "#ffd588",
    "#ffca66",
    "#ffbe44",
    "#ffb222",
    "#FFA600",
    "#dd9000",
    "#bb7a00",
    "#996400",
    "#774e00",
  ],
  red: [
    "#f4d1d1",
    "#edb6b6",
    "#e69b9b",
    "#DD7676",
    "#d96464",
    "#D24949",
    "#c83131",
    "#ac2b2b",
    "#912424",
    "#761d1d",
  ],
};
