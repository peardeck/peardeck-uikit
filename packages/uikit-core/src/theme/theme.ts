import colorNames from "./colors";

/**
 * https://styled-system.com/theme-specification
 */

const colors = {
  ...colorNames,
  background: colorNames.white,
  text: "#354242",
  primary: "#d24949",
  secondary: "#3087b7",
  accent: "#698600",
  pear: "#c9de55",
  muted: "#9B9B9B",
  border: colorNames.gray[5],
  link: {
    default: "#3087B7",
    hover: "#25698f",
    visited: "#3087B7",
    active: "#29abe2",
    focus: "#3087B7",
  },
};

const fontSizes = [16, 20, 24, 32, 48, 64, 72];
fontSizes[-1] = 14;
fontSizes[-2] = 11;
fontSizes[-3] = 8;

export const theme = {
  mediaQueries: {},

  // For some reason breakpoints don't work with numeric values?
  breakpoints: [450, 700, 900].map((size) => `${size}px`),
  space: [0, 4, 8, 16, 32, 64, 128, 256, 512],
  colors,
  fonts: {
    body: ["ProximaNovaSoft-Regular", "sans-serif"],
    heading: ["ProximaNovaSoft-SemiBold", "sans-serif"],
    monospace: [
      "Consolas",
      '"Liberation Mono"',
      "Menlo",
      "Courier",
      "monospace",
    ],
  },
  fontSizes,
  fontWeights: {
    light: 300,
    normal: 400,
    medium: 500,
    semibold: 600,
    bold: 700,
    heavy: 800,
  },
  lineHeights: {
    condensedUltra: 1,
    condensed: 1.25,
    default: 1.5,
    body: 1.7,
    title: 1.25,
  },
  letterSpacings: [],
  sizes: [],
  borders: [0, `1px solid ${colors.border}`],
  borderStyles: {},
  borderWidths: [],
  radii: [0, 3, 6, 12, 24, 48, "100%"],
  shadows: {
    [-2]: "rgba(0,0,0,0.12) 0px 2px 4px inset",
    [-1]: "rgba(0,0,0,0.12) 0px 1px 2px inset",
    0: "none",
    1: "0 1px 4px 0 rgba(0,0,0,0.27)",
    2: "0 2px 8px 0 rgba(0,0,0,0.27)",
    3: "0 4px 16px 0 rgba(0,0,0,0.27)",
    4: "0 8px 32px 0 rgba(0,0,0,0.27)",

    header: "0 1px 4px 0 rgba(0,0,0,0.27)",

    button: "0 1px 4px 0 rgba(0,0,0,0.27)",
    buttonHover: "0 2px 8px 0 rgba(0,0,0,0.27)",
    buttonActive: "rgba(0,0,0,0.12) 0px 0px 1px inset",

    // button: "0 2px 8px 0 rgba(0,0,0,0.27)",
    // buttonHover: "0 2px 8px 0 rgba(0,0,0,0.27)",
    // buttonActive: "",
    // buttonFocus: "",

    // 1: "0 1px 2px 0 rgba(0,0,0,0.4)",
    // 2: "0 2px 4px 0 rgba(0,0,0,0.4)",
    // 3: "0 4px 8px 0 rgba(0,0,0,0.4)",
    // 4: "0 8px 16px 0 rgba(0,0,0,0.4)",
    // formControl: "rgba(27, 31, 35, 0.075) 0px 1px 2px inset",
    // formControlFocus: "rgba(3, 102, 214, 0.3) 0px 0px 0px 0.2em",
    // glow: ``,
  },
  zIndices: [undefined, 0, 1, 2, 10, 100, 200, 10000],

  buttonColors: {
    none: {
      color: "text",
      backgroundColor: "transparent",
      hoverBackgroundColor: "transparent",
      borderColor: "transparent",
    },
    default: {
      color: "#000000",
      backgroundColor: "#ffffff",
      hoverBackgroundColor: "#f2f2f2",
      borderColor: "#cccccc",
    },
    primary: {
      color: "#ffffff",
      backgroundColor: "primary",
      hoverBackgroundColor: "#ba2e2e",
      borderColor: "#ffffff",
    },
    primaryOutline: {
      color: "primary",
      backgroundColor: "#ffffff",
      hoverBackgroundColor: "#eeeeee",
      borderColor: "#d24949",
      borderWidth: 2,
    },
    secondary: {
      color: "#ffffff",
      backgroundColor: "secondary",
      hoverBackgroundColor: "#2b78a3",
      borderColor: "#ffffff",
    },
    secondaryOutline: {
      color: "secondary",
      backgroundColor: "#ffffff",
      hoverBackgroundColor: "#eeeeee",
      borderColor: "#3087b7",
      borderWidth: 2,
    },
    accent: {
      color: "#ffffff",
      backgroundColor: "accent",
      hoverBackgroundColor: "#556d00",
      borderColor: "#ffffff",
    },
    accentOutline: {
      color: "accent",
      backgroundColor: "#ffffff",
      hoverBackgroundColor: "#eeeeee",
      borderColor: "#698600",
      borderWidth: 2,
    },
  },

  buttonStyles: {
    default: {
      borderRadius: 2,
      borderWidth: 1,
      boxShadow: 1,
      fontWeight: "bold",
    },
    callToAction: {
      borderRadius: 4,
      borderWidth: 2,
      boxShadow: 1,
      fontWeight: "bold",
    },
  },

  colorStyles: {
    success: {
      color: "#8FA340",
    },
    info: {
      color: "#444546",
    },
    danger: {
      color: "#D24949",
    },
    warning: {
      color: "#F5E98F",
    },
  },

  textStyles: {
    caps: {
      textTransform: "uppercase",
    },
    smallCaps: {
      textTransform: "uppercase",
      letterSpacing: "0.1em", // TODO: Replace with `letterSpacings` value
    },
  },
};

export type Theme = typeof theme;

declare module "styled-components" {
  // eslint-disable-next-line @typescript-eslint/no-empty-interface
  export interface DefaultTheme extends Theme {}
}

export interface ThemedComponentProps {
  theme?: Theme;
}
