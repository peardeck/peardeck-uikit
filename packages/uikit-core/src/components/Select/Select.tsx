import styled from "styled-components";
import {
  color,
  ColorProps,
  display,
  DisplayProps,
  space,
  SpaceProps,
  typography,
  TypographyProps,
} from "styled-system";
import { theme, ThemedComponentProps } from "../../theme/theme";

export type SelectProps = ThemedComponentProps &
  TypographyProps &
  ColorProps &
  DisplayProps &
  SpaceProps;

/**
 * Select is component for setting font-size, typographic styles, margin, and color.
 */
export const Select = styled.select<SelectProps>`
  ${color}
  ${display}
  ${space}
  ${typography}
`;

Select.defaultProps = { theme, p: 1 };

Select.displayName = "Select";
