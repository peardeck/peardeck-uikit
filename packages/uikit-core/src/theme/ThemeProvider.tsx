import React, { ReactNode } from "react";
import { ThemeProvider as StyledThemeProvider } from "styled-components";
import { ThemedComponentProps, theme as defaultTheme } from "./theme";

export interface ThemeProviderProps extends ThemedComponentProps {
  children: ReactNode;
}

// TODO: Add global styles/reset.

export const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps): JSX.Element => (
  <StyledThemeProvider theme={theme}>
    <>{children}</>
  </StyledThemeProvider>
);
