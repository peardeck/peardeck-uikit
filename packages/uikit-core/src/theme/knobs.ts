import { number, select } from "@storybook/addon-knobs";
import { theme } from "./theme";

/**
 * This contains various predefined knobs and helper methods for creating
 * knobs.
 *
 * Notes:
 *
 * * The `createOptions...` methods are used to generated options from existing
 *   scales or arrays more easily, while additionally adding a "not specified"
 *   option to select, which effect means "the default value when the prop
 *   is not given"
 *
 * * The `select` knob does not display options with an underlying value of
 *   `undefined` - this is why the default and "not specified" values are
 *   actually set to `null`. Then each knob uses the `nullToUndefined`
 *   helper to convert it correctly to undefined.
 */

export const GROUP_ID_TYPOGRAPHY = "Typography";
export const GROUP_ID_COLOR = "Color";
export const GROUP_ID_SPACE = "Space";
export const GROUP_ID_DISPLAY = "Display";
export const GROUP_ID_BORDER = "Border";
export const GROUP_ID_LAYOUT = "Layout";

/* *************************************
 * Helper methods
 ************************************ */

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const nullToUndefined = (result: any) => (result === null ? undefined : result);

/**
 * Creates options for a `select` knob that also includes a "not specified"
 * value, with the option labels matching the array _values_.
 * @param values An array of values
 */
export const createOptionsFromArray = (values: (string | number)[]) =>
  values.reduce((options, value) => ({ ...options, [value]: value }), {
    "not specified": null,
  });

/**
 * Creates options for a `select` knob that also includes a "not specified"
 * value, with the option labels matching the array _indexes_.
 * @param values An array of values
 */
export const createOptionsFromScale = (values: (string | number)[]) =>
  values.reduce((options, _, index) => ({ ...options, [index]: index }), {
    "not specified": null,
  });

/**
 * Creates options for a `select` knob that also includes a "not specified"
 * value, with the option labels generated using the keys of the map
 * @param map A map of values
 */
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const createOptionsFromMap = (map: { [key: string]: any }) =>
  Object.keys(map).reduce((options, key) => ({ ...options, [key]: key }), {
    "not specified": null,
  });

/* *************************************
 * Predefined knobs
 ************************************ */

/**
 * A knob for selecting a font size value from the theme-defined scale of font sizes scale
 */
export const themeFontSizeKnob = (label = "Font size", initial = null) =>
  nullToUndefined(
    select(
      label,
      createOptionsFromScale(theme.fontSizes),
      initial,
      GROUP_ID_TYPOGRAPHY
    )
  );

/**
 * A knob for selecting a font family value from the theme-defined font families
 */
export const themeFontFamilyKnob = (label = "Font Family", initial = null) =>
  nullToUndefined(
    select(
      label,
      createOptionsFromMap(theme.fonts),
      initial,
      GROUP_ID_TYPOGRAPHY
    )
  );

/**
 * A knob for selecting a font weight value from the theme-defined font weights
 */
export const themeFontWeightKnob = (label = "Font Weight", initial = null) =>
  nullToUndefined(
    select(
      label,
      createOptionsFromMap(theme.fontWeights),
      initial,
      GROUP_ID_TYPOGRAPHY
    )
  );

/**
 * A knob for selecting a line height value from the theme-defined line heights
 */
export const themeLineHeightKnob = (label = "Line Height", initial = null) =>
  nullToUndefined(
    select(
      label,
      createOptionsFromMap(theme.lineHeights),
      initial,
      GROUP_ID_TYPOGRAPHY
    )
  );

/**
 * A knob for selecting a color value from the theme-defined (named) colors
 * (For _any_ color, use the `color` knob provided by `@storybook/addon-knobs`)
 */
export const themeColorKnob = (
  label = "Color",
  initial: null | string = null
) =>
  nullToUndefined(
    select(label, createOptionsFromMap(theme.colors), initial, GROUP_ID_COLOR)
  );

/**
 * A knob for selecting an opacity value from 0 to 1.
 */
export const opacityKnob = (label = "Opacity", initial = 1) =>
  nullToUndefined(
    number(
      label,
      initial,
      { range: true, min: 0, max: 1, step: 0.01 },
      GROUP_ID_COLOR
    )
  );

/**
 * A knob for selecting a letter spacing value from the theme-defined letter spacings scale
 */
export const themeLetterSpacingKnob = (
  label = "Letter Spacing",
  initial = null
) =>
  nullToUndefined(
    select(
      label,
      createOptionsFromScale(theme.letterSpacings),
      initial,
      GROUP_ID_SPACE
    )
  );

/**
 * A knob for selecting a border value from the theme-defined borders
 */
export const themeBordersKnob = (label = "Border", initial = null) =>
  nullToUndefined(
    select(
      label,
      createOptionsFromScale(theme.borders),
      initial,
      GROUP_ID_BORDER
    )
  );

/**
 * A knob for selecting a radius value from the theme-defined radii scale
 */
export const themeRadiiKnob = (label = "Border Radius", initial = null) =>
  nullToUndefined(
    select(label, createOptionsFromScale(theme.radii), initial, GROUP_ID_BORDER)
  );

/**
 * A knob for selecting a shadow value from the theme-defined shadows scale
 */
export const themeShadowsKnob = (
  label = "Shadow",
  initial: number | null = null
) =>
  nullToUndefined(
    select(
      label,
      createOptionsFromScale(theme.shadows),
      initial,
      GROUP_ID_BORDER
    )
  );

/**
 * A knob for selecting a spacing value from the theme-defined space scale
 */
export const themeSpaceKnob = (label = "Spacing", initial = null) =>
  nullToUndefined(
    select(label, createOptionsFromScale(theme.space), initial, GROUP_ID_SPACE)
  );

/**
 * A knob for selecting a standard font style value
 */
export const fontStyleKnob = (label = "Font Style", initial = null) =>
  nullToUndefined(
    select(
      label,
      createOptionsFromArray(["normal", "italic", "oblique"]),
      initial,
      GROUP_ID_TYPOGRAPHY
    )
  );

/**
 * A knob for selecting a standard text alignment attribute value
 */
export const textAlignKnob = (label = "Text Align", initial = null) =>
  nullToUndefined(
    select(
      label,
      createOptionsFromArray([
        "inherit",
        "initial",
        "revert",
        "unset",
        "center",
        "end",
        "justify",
        "left",
        "right",
        "start",
      ]),
      initial,
      GROUP_ID_TYPOGRAPHY
    )
  );

/**
 * A knob for selecting a standard display attribute value
 */
export const displayKnob = (label = "Display", initial = null) =>
  nullToUndefined(
    select(
      label,
      createOptionsFromArray([
        "block",
        "inline",
        "run-in",
        "inline-block",
        "flex",
        "table",
        "grid",
        "list-item",
        "none",
      ]),
      initial,
      GROUP_ID_DISPLAY
    )
  );

export const elementTypes = ["span", "div", "p"];

/**
 * A knob for selecting an element type (for an "as" prop) from a predefined
 * list of element types. (feel free to add additional types here if needed)
 */
export const elementTypeKnob = (label = "Element Type", initial = null) =>
  nullToUndefined(
    select(label, createOptionsFromArray(["span", "div", "p"]), initial)
  );

/* *************************************
 * Knob groups
 ************************************ */

/**
 * A collection of knobs for changing topography-related values.
 */
export const typographyKnobs = () => ({
  fontFamily: themeFontFamilyKnob(),
  fontSize: themeFontSizeKnob(),
  fontWeight: themeFontWeightKnob(),
  lineHeight: themeLineHeightKnob(),
  letterSpacing: themeLetterSpacingKnob(),
  fontStyle: fontStyleKnob(),
  textAlign: textAlignKnob(),
});

/**
 * Creates a collection of knobs for changing space-related values.
 */
export const spaceKnobs = () => ({
  m: themeSpaceKnob("Margin"),
  p: themeSpaceKnob("Padding"),
});

/**
 * Creates a collection of knobs for changing color-related values.
 */
export const colorKnobs = () => ({
  color: themeColorKnob("Foreground Color"),
  backgroundColor: themeColorKnob("Background Color"),
  opacity: opacityKnob(),
});

/**
 * Creates a collection of knobs for changing display-related values.
 */
export const displayKnobs = () => ({
  display: displayKnob(),
});

export const borderKnobs = () => ({
  border: themeBordersKnob(),
  boxShadow: themeShadowsKnob("Box Shadow"),
  borderRadius: themeRadiiKnob("Border Radius"),
});

export const layoutKnobs = () => ({
  // width,
  // height,
  // minWidth,
  // minHeight,
  // maxWidth,
  // maxHeight,
  // size,
  // overflow,
  // display,
  // verticalAlign,
});
