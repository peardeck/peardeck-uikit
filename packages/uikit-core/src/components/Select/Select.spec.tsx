import "jest-styled-components";
import { create } from "react-test-renderer";
import React from "react";
import { Select } from "./Select";
import { theme } from "../../theme/theme";

describe("Select", () => {
  it("handles `fontSize` values from theme", () => {
    const json = create(<Select fontSize={2} />).toJSON();
    expect(json).toMatchSnapshot();
    expect(json).toHaveStyleRule("font-size", `${theme.fontSizes[2]}px`);
  });

  it("allows non-theme `fontSize` values", () => {
    const json = create(<Select fontSize="200px" />).toJSON();
    expect(json).toMatchSnapshot();
    expect(json).toHaveStyleRule("font-size", "200px");
  });

  it("handles `fontWeight` values from theme", () => {
    const json = create(<Select fontWeight="normal" />).toJSON();
    expect(json).toMatchSnapshot();
    expect(json).toHaveStyleRule(
      "font-weight",
      String(theme.fontWeights.normal)
    );
  });

  it("handles `fontStyle` values", () => {
    const json = create(<Select fontStyle="italic" />).toJSON();
    expect(json).toMatchSnapshot();
    expect(json).toHaveStyleRule("font-style", "italic");
  });

  it("handles `lineHeight` values from theme", () => {
    const json = create(<Select lineHeight="condensedUltra" />).toJSON();
    expect(json).toMatchSnapshot();
    expect(json).toHaveStyleRule(
      "line-height",
      String(theme.lineHeights.condensedUltra)
    );
  });

  it('respects fontFamily="monospace"', () => {
    const json = create(<Select fontFamily="monospace" />).toJSON();
    expect(json).toMatchSnapshot();
    expect(json).toHaveStyleRule(
      "font-family",
      theme.fonts.monospace.join(",")
    );
  });
});
