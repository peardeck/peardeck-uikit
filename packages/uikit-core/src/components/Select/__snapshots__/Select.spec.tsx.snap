// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select allows non-theme \`fontSize\` values 1`] = `
.c0 {
  padding: 4px;
  font-size: 200px;
}

<select
  className="c0"
  fontSize="200px"
/>
`;

exports[`Select handles \`fontSize\` values from theme 1`] = `
.c0 {
  padding: 4px;
  font-size: 24px;
}

<select
  className="c0"
  fontSize={2}
/>
`;

exports[`Select handles \`fontStyle\` values 1`] = `
.c0 {
  padding: 4px;
  font-style: italic;
}

<select
  className="c0"
  fontStyle="italic"
/>
`;

exports[`Select handles \`fontWeight\` values from theme 1`] = `
.c0 {
  padding: 4px;
  font-weight: 400;
}

<select
  className="c0"
  fontWeight="normal"
/>
`;

exports[`Select handles \`lineHeight\` values from theme 1`] = `
.c0 {
  padding: 4px;
  line-height: 1;
}

<select
  className="c0"
/>
`;

exports[`Select respects fontFamily="monospace" 1`] = `
.c0 {
  padding: 4px;
  font-family: Consolas,"Liberation Mono",Menlo,Courier,monospace;
}

<select
  className="c0"
  fontFamily="monospace"
/>
`;
